bin/check-bldr-toml.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+#
+# pre-commit hook:
+# Check for plans that don't have an entry in our .bldr.toml
+#
+# Authors:
+# - Steven Danna <steve@chef.io>
+
+set -eu
+requested=($@)
+
+check_bldr_toml_for() {
+  plan_path=$1
+  # Alternatively, we could do
+  #     plan_name=$(dirname $plan_path)
+  # if we think directory names will always match plan names.
+  plan_name=$(bash -c "source $plan_path; echo \"\$pkg_name\"")
+  if ! grep -Eq "^\[$plan_name\]" .bldr.toml
```

Comment:
Is this grep syntax universal? How might double quotes affect the escaping of the expression?

Reply:
I think this should be OK. The options `-q` and `-E` are both POSIX options. The "^" as a begin of line anchor is standard and "[" is a special character in both regular and extended regular expressions so it needs to be escaped.  We don't technically need the -E, I just included it since this was modified from another one of our check scripts:

- http://pubs.opengroup.org/onlinepubs/009695399/utilities/grep.html
- http://pubs.opengroup.org/onlinepubs/009695399/basedefs/xbd_chap09.html#tag_09_04

> How might double quotes affect the escaping of the expression?

Good question! But I think we are OK here too.  The double quotes don't effect the \[ here because, according to `man bash`:

>  Enclosing characters in double quotes preserves the literal value of all characters within the quotes, with the exception of $, `, \, and, when history expansion  is  enabled,  !.

That is,

```
> echo "\$foo"
$foo
> echo "\[foo"
\[foo
```
 I /think/ this should be OK.  If we move away from bash, we likely need to change a number of these scripts.